### PR TITLE
notify DC_EVENT_INCOMING_WEBXDC_NOTIFY

### LINKED ISSUE #3439: add webxdc selfAddr/notify/href APIs (M2/M3)

### PR DESCRIPTION
**Pull request by adbenitez**
close #3439

### CODE REVIEW

File: src/main/java/org/thoughtcrime/securesms/notifications/NotificationCenter.java [hunk 1]
```diff
@@ -366,7 +368,28 @@ public void notifyReaction(int accountId, int contactId, int msgId, String react
 
         DcContact sender = dcContext.getContact(contactId);
         String shortLine = context.getString(R.string.reaction_by_other, sender.getDisplayName(), reaction, dcMsg.getSummarytext(2000));
-        maybeAddNotification(accountId, dcContext.getChat(dcMsg.getChatId()), msgId, shortLine, shortLine, false);
+        maybeAddNotification(accountId, dcContext.getChat(dcMsg.getChatId()), msgId, shortLine, shortLine, true);
```
Comment (r10s) [inline src/main/java/org/thoughtcrime/securesms/notifications/NotificationCenter.java]: the "play in chat sound" needs to be false here and for webxdc-info.

at some point, we can factor that out , but well ...

**Review by r10s — COMMENTED** [review]
otherwise lgtm

File: src/main/java/org/thoughtcrime/securesms/notifications/NotificationCenter.java [hunk 2]
```diff
@@ -370,6 +372,27 @@ public void notifyReaction(int accountId, int contactId, int msgId, String react
...
+          return; // showing "New Message" is wrong, just do nothing.
+        }
+
+        DcContext dcContext = context.dcAccounts.getAccount(accountId);
+        DcMsg dcMsg = dcContext.getMsg(msgId);
+        JSONObject info;
+        if(dcMsg.getType() == DcMsg.DC_MSG_WEBXDC) {
+          info = dcMsg.getWebxdcInfo();
+        } else { // info message, get from parent xdc
+          info = dcMsg.getParent().getWebxdcInfo();
```
Comment (r10s) [inline src/main/java/org/thoughtcrime/securesms/notifications/NotificationCenter.java:388]: getParent() can be null, so maybe be more resilient here, a try/catch around things or so

**Review by r10s — APPROVED** [review]

**Comment by adbenitez** [comment]
there are some problems I see: since if the notification comes from a summary change, there is no actual message id that is different per notification, several notifications like that for the same message would probably override each other since they will have the same id

we would need to generate some new notifications ID, and handle them differently also to be able to discard notifications of a webxdc when you open it (similar to how we discard notifications of a chat when you open it)

not notifying if the notification is from the currently opened webxdc is more easy to do

**Review by r10s — APPROVED** [review]

**Pull request merged**